ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs

```diff
@@ -0,0 +1,593 @@
+
+                    // Add the Cnl object to the list
+                    selectedChannels.Add(cnl);
+                }
+            }
+
+            // Use the AddChannels function to add the selected channels
+            AddChannels(selectedChannels);
+        }
+
+
+        private List<Cnl> CreateChannelsFromFille(Dictionary<string, List<string>> store)
+        {
+            List<Cnl> cnls = new List<Cnl>();
+            int cnlNum = CtrlCnlImport3.StartCnlNum;
+            string namePrefix = CtrlCnlImport1.SelectedDevice.Name;
+            int? objNum = CtrlCnlImport2.ObjNum;
+            int deviceNum = CtrlCnlImport1.SelectedDevice.DeviceNum;
+
+            foreach (KeyValuePair<string, List<string>> element in store)
+            {
+                string tagCode = element.Key;
+                CnlPrototype cnlPrototype = CtrlCnlImport1.CnlPrototypes.FirstOrDefault(); // or some other way to get the prototype
+
+                if (cnlPrototype != null)
+                {
+                    cnls.Add(new Cnl
+                    {
+                        CnlNum = cnlNum,
+                        Active = cnlPrototype.Active,
+                        Name = namePrefix + cnlPrototype.Name,
+                        DataTypeID = cnlPrototype.DataTypeID,
+                        DataLen = cnlPrototype.DataLen,
+                        CnlTypeID = cnlPrototype.CnlTypeID,
+                        ObjNum = objNum,
+                        DeviceNum = deviceNum,
+                        TagNum = cnlPrototype.TagNum,
+                        TagCode = tagCode,
+                        FormulaEnabled = cnlPrototype.FormulaEnabled,
+                        InFormula = cnlPrototype.InFormula,
+                        OutFormula = cnlPrototype.OutFormula,
+                        FormatID = project.ConfigDatabase.GetFormatByCode(cnlPrototype.FormatCode)?.FormatID,
+                        QuantityID = project.ConfigDatabase.GetQuantityByCode(cnlPrototype.QuantityCode)?.QuantityID,
+                        UnitID = project.ConfigDatabase.GetUnitByCode(cnlPrototype.UnitCode)?.UnitID,
+                        LimID = null,
+                        ArchiveMask = cnlPrototype.ArchiveMask,
+                        EventMask = cnlPrototype.EventMask
+                    });
+
+                    cnlNum++; // Increment cnlNum for each element in store
+                }
+            }
+
+            return cnls;
+        }
+        public Dictionary<string, List<string>> StoreDictioData()
+        {
+            Dictionary<string, List<string>> resultDictionnary = new Dictionary<string, List<string>>();
+
+            // Parcourir dictio et ajouter des éléments à resultDictionnary
+            foreach (KeyValuePair<string, List<string>> kvp in dictio)
+            {
+                var list = new List<string>
+        {
+            kvp.Value[0], // Mnemonique
+            kvp.Value[1], // Type
+            kvp.Value[2]  // Descr
```

Comment:
Prefer adding a single comment before the function than comment every line, except specific need